src/south/SouthHandler.class.js

```diff
@@ -34,45 +34,38 @@ const COMPRESSION_LEVEL = 9
  * handled by the OIBus engine and should not be taken care at the South level.
  *
  */
-class ProtocolHandler {
+class SouthHandler {
   /**
    * Constructor for Protocol
    * @constructor
    * @param {*} dataSource - The data source
    * @param {BaseEngine} engine - The engine
-   * @param {object} supportedModes - The supported modes
+   * @param {Object} schema - The schema defining this South
    * @return {void}
    */
-  constructor(dataSource, engine, supportedModes) {
+  constructor(dataSource, engine) {
     this.dataSource = dataSource
     this.engine = engine
+    const schema = engine.southSchemas[dataSource.name]
     this.encryptionService = EncryptionService.getInstance()
     const { engineConfig } = this.engine.configService.getConfig()
     this.engineConfig = engineConfig
-    this.supportedModes = supportedModes
-
     this.connected = false
-
     this.addFileCount = 0
     this.addPointsCount = 0
     this.currentlyOnScan = {}
     this.buffer = []
     this.bufferTimeout = null
     this.statusData = {}
-
     this.keyFile = null
     this.certFile = null
     this.caFile = null
+    this.supportedModes = this.checkSupportedModes(schema)
   }
 
-  checkSupportedModes() {
-    const {
-      supportListen,
-      supportLastPoint,
-      supportFile,
-      supportHistory,
-    } = this.supportedModes
-    if (!supportListen && !supportLastPoint && !supportFile && !supportHistory) {
+  checkSupportedModes(schema) {
+    const { supportListen, supportLastPoint, supportPoints, supportFiles, supportHistory } = schema
+    if (!supportListen && !supportLastPoint && !supportFiles && !supportHistory) {
```

Comment:
Should we use false default values for these props, to be consistent between false, undefined or null ?